pkg/app/app_test.go

```diff
@@ -39,6 +39,12 @@ func appWithFs(app *App, files map[string]string) *App {
 }
 
 func injectFs(app *App, fs *testhelper.TestFs) *App {
+	if app.Set == nil {
```

Comment:
Didn't really want to add this in `injectFs`, but the alternatives seemed to be either adding it to every test case, or doing a refactor of the tests usage of `App` at the same time. As such, I preferred to keep this change simple, and the appropriate refactoring can be discussed and addressed afterwards?

Note that there are several cases where tests setup `App` and _do not_ call this function, however those tests all directly call `app.loadDesiredStateFromYaml`. While that could _would_ be impacted by the change to `app.Set` in real use, in the tests no options are provided to the call, so any issues aren't exposed there.